packages/yoastseo/src/scoring/assessor.js

```diff
@@ -120,6 +124,9 @@ Assessor.prototype.getMarker = function( assessment, paper, researcher ) {
 Assessor.prototype.assess = function( paper ) {
 	this._researcher.setPaper( paper );
 
+	const languageProcessor = new LanguageProcessor( this._researcher );
+	paper.setTree( filterTree( build( paper.getText(), languageProcessor ), permanentFilters ) );
```

Comment:
Could you explain (in technical) choices why you are re-setting and re-building the tree here? As this also happens in the analysisWebworker.

Reply:
The same change is also made in this PR: https://github.com/Yoast/wordpress-seo/pull/20070.
This step is to make sure that we rebuild the tree when we switch the assessor to analyze the related keyphrase. The changed is also made here so that in this PR we can also test the related keyphrase.